bundles/org.openhab.transform.basicprofiles/src/main/java/org/openhab/transform/basicprofiles/internal/profiles/TimeweightedAverageStateProfile.java

```diff
@@ -0,0 +1,241 @@
+
+    public TimeweightedAverageStateProfile(ProfileCallback callback, ProfileContext context) {
+        this.callback = callback;
+        this.scheduler = context.getExecutorService();
+        this.config = context.getConfiguration().as(TimeweightedAverageProfileConfig.class);
+        itemName = callback.getItemChannelLink().getItemName();
+        try {
+            scheduleDuration = DurationUtils.parse(config.duration);
+        } catch (IllegalArgumentException e) {
+            scheduleDuration = DEFAULT_TIMEOUT;
+            logger.warn("Invalid duration configuration {} for item {}. Fallback to {}", config.duration, itemName,
+                    DEFAULT_TIMEOUT);
+        }
+    }
+
+    @Override
+    public ProfileTypeUID getProfileTypeUID() {
+        return TIME_WEIGHTED_AVERAGE_UID;
+    }
+
+    @Override
+    public void onStateUpdateFromHandler(State state) {
+        logger.trace("Received {} {}", itemName, state);
+
+        // first call initialization and threshold check
+        // synchronize access to timeframe and latestState to avoid parallel execution of delivery by twaJob
+        synchronized (timeframe) {
+            if (latestState == null) {
+                init(state);
+            }
+
+            // if state change is above delta threshold, deliver immediately collected values plus latest reported state
+            if (deltaExceeded(state)) {
+                deliver();
+                callback.sendUpdate(state);
+            }
+
+            // start new time frame
+            startJob();
+            timeframe.put(Instant.now(), state);
+            latestState = state;
+            streamingInTimeframe = true;
+        }
+    }
+
+    private void init(State first) {
+        if (first instanceof QuantityType<?> qtState) {
+            stateUnit = qtState.getUnit();
+            logger.debug("Profile initialized for {} with QuantityType {}", itemName, stateUnit);
+        } else if (first instanceof DecimalType) {
+            // nothing to do, stateUnit stays empty
+            logger.debug("Profile initialized for {} with DecimalType", itemName);
+        } else {
+            logger.error("Time-weighted average profile not applicable for {} with class {}", itemName,
+                    first.getClass());
+        }
+    }
+
+    private boolean deltaExceeded(State state) {
+        if (config.delta > 0) {
+            State localLatestState = latestState;
+            if (localLatestState != null) {
+                double delta = Math.abs(state2Double(state) - state2Double(localLatestState));
+                if (delta >= config.delta) {
+                    logger.debug("{} rapid change from {} to {}", itemName, latestState, state);
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    private void deliver() {
+        TreeMap<Instant, State> delivery = prepareDelivery();
+        if (delivery.size() <= 1) {
```

Comment:
When the data >= delta, I presume that delivery.size() would be > 0, hence it won't satisfy this condition?